src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,365 @@
+            "dtype": dtype,
+            "values": initial_values,
+        },
+        "attributes": [],
+    }
+    if unit is not None:
+        attrs = dataset["attributes"]
+        attrs.append(
+            {
+                "name": "units",
+                "dtype": "string",
+                "values": unit,
+            }
         )
-        group['children'].append(dataset)
     return dataset
 
 
-def _merge_ev44(group, message):
-    for field, value in message.items():
-        if value is None:  # Could be monitor without pixel_id
-            continue
-        dataset = _init_dataset(group, field, value)
-        if field in ('time_of_flight', 'pixel_id'):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
-            )
+ModuleNameType = Literal["ev44"]  # "f144", "tdct" will be added in the future
+"""Name of the module that is supported by beamlime."""
+
+
+def _get_array_values(dataset: Mapping) -> np.ndarray:
+    return dataset["config"]["values"]
+
+
+def _set_values(dataset: Mapping, values: Any) -> None:
+    dataset["config"]["values"] = values
+
 
+def _initialize_ev44(group: DataModuleParent) -> None:
+    """Initialize ev44 datasets in the parent.
 
-def _merge_f144(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamp', 'value'):
-            dataset['config']['values'] = np.stack((dataset['config']['values'], value))
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
 
+    Side Effects
+    ------------
+    ``children`` of the ``group`` will have 4 datasets with empty values.
 
-def _merge_tdct(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamps',):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
+    - event_time_zero
+    - event_time_offset
+    - event_index
+    - event_id
+
+    This function doesn't remove a place holder from the ``children``.
+    It is expected to be removed outside of this function.
+
+    """
+    children = group.setdefault("children", [])
+    # event_time_zero
+    children.append(
+        create_dataset(
+            name="event_time_zero",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_time_offset
+    children.append(
+        create_dataset(
+            name="event_time_offset",
+            dtype="int",
+            initial_values=np.asarray([]),
+            unit="ns",
+        )
+    )
+    # event_index
+    children.append(
+        create_dataset(
+            name="event_index",
+            dtype="int",
+            initial_values=np.asarray([]),
+        )
+    )
+    # event_id
+    if "monitor" in group["name"]:
```

Comment:
Isn't it better to look at `data_part` and only set `event_id` if `pixel_id` is not `None`?

Reply:
I thought `initialize` function shouldn't depend on the data it accepts.
But I looked at other schemas and there are quite a few optional fields as well... so I'll do that.

Reply:
Hm good point. Not sure what's best. Maybe we can use the first message in the initialization to determine the optional fields, but not populate the fields with data.